src/graphics/webgpu/webgpu-bind-group-format.js

```diff
@@ -0,0 +1,141 @@
+/** @typedef {import('../bind-group-format.js').BindGroupFormat} BindGroupFormat */
+/** @typedef {import('./webgpu-graphics-device.js').WebgpuGraphicsDevice} WebgpuGraphicsDevice */
+
+import { WebgpuUtils } from './webgpu-utils.js';
+
+/**
+ * A WebGPU implementation of the BindGroupFormat, which is a wrapper over GPUBindGroupLayout.
+ *
+ * @ignore
+ */
+class WebgpuBindGroupFormat {
+    /**
+     * Unique key, used for caching
+     *
+     * @type {string}
+     */
+    key = '';
+
+    /**
+     * @param {BindGroupFormat} bindGroupFormat -
+     */
+    constructor(bindGroupFormat) {
+
+        /** @type {WebgpuGraphicsDevice} */
+        const device = bindGroupFormat.device;
```

Comment:
wondering, is this line required to basically typecast device to webgpu-specific device to circumvent error checking? it's very clunky syntax.

Reply:
it is, and agreed, but I'm not sure JS has better syntax for types.